app/controllers/posts_controller.rb

```diff
@@ -24,6 +24,21 @@ def create
     end
   end
 
+  def destroy
+    @user = User.find(params[:user_id])
+    @post = @user.posts.find(params[:id])
+    @post.likes.destroy_all
+    @post.comments.destroy_all
+    @post.destroy
+    if @post.destroy
+      flash[:notice] = 'Post deleted !'
+      redirect_to user_posts_path(@user)
+    else
+      flash[:alert] = 'The post could not be deleted !'
+      redirect_to user_post_path(@user)
+    end
+  end
```

Comment:
- [ ] Great jos adding authorization rules to your app. At the start of your postsController, you're required use CanCanCan to add posts authorization in Post Controller. Kindly add that, It should look something similar to this :point_down: 

```
   class PostsController < ApplicationController
    load_and_authorize_resource

    ...
  end

```

- Note: Kindly ensure this is added at the beginning of your controller before any actions at all. I could not highlight that part of the because, no chages was made there. So I decided to highlight here that could be highlighted. Please add it to the top of your controller.

Reply:
Addressed